### DISCUSSION TITLE
Can we disable entire chain authentication

### DISCUSSION THREAD

**Discussion by amvarunn** [strongswan/strongswan#884] [Q&A]
Consider a scenario where we use certificate chains with CA1Root, CA1intermediate1, CA1intermediate2 and CA1intermediate3 on Primary, and CA2Root, CA2intermediate1, CA2intermediate2 and CA2intermediate3 on remote side. 
End entity certificates are generated using the respective intermediate3 certificates on both sides.

Is it always required to have all certs of the Primay cert chain, in remote site?
Can we only have CA1intermediate3 which signed the Primary end entity, in the remote side and create tunnel?

Example:
I have tried with Primary site having its end entity certificate + private key, and remote site having its own end entity, private Key and IntermediateCA3 which signed the end point of Primary

`Primary/Initiator

Feb  7 01:18:46 13[CFG] loaded 0 entries for attr plugin configuration
Feb  7 01:18:47 07[CFG] loaded certificate 'C=x, O=y, CN=device1'
Feb  7 01:18:47 16[CFG] loaded RSA private key
Feb  7 01:18:47 10[CFG] updated vici connection: host-host
Feb  7 01:18:54 10[CFG] vici initiate IKE_SA 'host-host'
Feb  7 01:18:54 14[IKE] <host-host|15> initiating IKE_SA host-host[15] to 192.168.204.212
Feb  7 01:18:54 08[CFG] <host-host|15> selected proposal: IKE:AES_CBC_128/HMAC_SHA2_256_128/PRF_AES128_XCBC/ECP_256
Feb  7 01:18:54 08[IKE] <host-host|15> received cert request for "C=y, O=x, CN=Root1 CA"
Feb  7 01:18:54 08[IKE] <host-host|15> received cert request for "C=y, O=x, CN=Root1 CA"
Feb  7 01:18:54 08[IKE] <host-host|15> received 1 cert requests for an unknown ca
Feb  7 01:18:54 08[IKE] <host-host|15> sending cert request for "C=y, O=x, CN=Root1 CA"
Feb  7 01:18:54 08[IKE] <host-host|15> sending cert request for "C=y, O=x, CN=Root1 CA"
Feb  7 01:18:54 08[IKE] <host-host|15> authentication of 'device1' (myself) with RSA_EMSA_PKCS1_SHA2_256 successful
Feb  7 01:18:54 08[IKE] <host-host|15> sending end entity cert "C=y, O=x, CN=device1"
Feb  7 01:18:54 14[IKE] <host-host|15> received AUTHENTICATION_FAILED notify error

Remote/Responder
Mon Feb  7 01:18:50 2022 02[CFG] loaded 0 entries for attr plugin configuration
Mon Feb  7 01:18:51 2022 15[CFG] loaded certificate 'C=y, O=x, CN=device2'
Mon Feb  7 01:18:51 2022 09[CFG] loaded certificate 'C=y, O=x, CN=intermediate3 CA'
Mon Feb  7 01:18:51 2022 01[CFG] updated vici connection: host-host
Mon Feb  7 01:18:54 2022 02[IKE] <15> 192.168.204.232 is initiating an IKE_SA
Mon Feb  7 01:18:54 2022 02[CFG] <15> selected proposal: IKE:AES_CBC_128/HMAC_SHA2_256_128/PRF_AES128_XCBC/ECP_256
Mon Feb  7 01:18:54 2022 02[IKE] <15> sending cert request for "C=y, O=x, CN=intermediate3 CA"
Mon Feb  7 01:18:54 2022 02[IKE] <15> sending cert request for "C=y, O=x, CN=Root1 CA"
Mon Feb  7 01:18:54 2022 02[IKE] <15> sending cert request for "C=y, O=x, CN=Root1 CA"
Mon Feb  7 01:18:54 2022 07[IKE] <15> received cert request for "C=y, O=x, CN=Root1 CA"
Mon Feb  7 01:18:54 2022 07[IKE] <15> received cert request for "C=y, O=x, CN=Root1 CA"
Mon Feb  7 01:18:54 2022 07[IKE] <15> received end entity cert "C=y, O=x, CN=device1"
Mon Feb  7 01:18:54 2022 07[CFG] <15> looking for peer configs matching 192.168.204.212[device2]...192.168.204.232[device1]
Mon Feb  7 01:18:54 2022 07[CFG] <host-host|15> selected peer config 'host-host'
Mon Feb  7 01:18:54 2022 07[CFG] <host-host|15>   using certificate "C=y, O=x, CN=device1"
Mon Feb  7 01:18:54 2022 07[CFG] <host-host|15> no issuer certificate found for "C=y, O=x, CN=device1"
Mon Feb  7 01:18:54 2022 07[CFG] <host-host|15>   issuer is "C=y, O=x, CN=Root1 CA"
Mon Feb  7 01:18:54 2022 07[IKE] <host-host|15> no trusted RSA public key found for 'device1'`

**Comment by tobiasbrunner** [comment]
> Can we only have CA1intermediate3 which signed the Primary end entity, in the remote side and create tunnel?

Only if that's the only trusted certificate you have installed from that chain, if you have the root certificate installed and receive a certificate request for it, it will act as trust anchor and all intermediate CAs have to be available. You know that the intermediate CA certificates can be sent by the peer? (That is, you only need the respective root CA installed locally.) And sending certificate requests could also be disabled.

But something in that log really doesn't look right. There are multiple certificate requests (sent and received) for the same root CA but then it fails to find that certificate while authenticating (at least one with the same subject DN). I suggest you fix your PKI and check other locations for installed certificates (e.g. if you also have stuff in `/etc/ipsec.d` loaded by the _stroke_ plugin, which you probably don't want if you use swanctl/vici).

**Reply by amvarunn** [reply]
Hi, Thanks for the above.
I perhaps wrongly phrased my question.

I am trying to find out, if there is a way in strongswan to prevent validation of entire certificate chain.

_Scenario: [NOTE: The certs in boldtext are not going to be installed anywhere]
step 1: Generate **RootCA1 -> primaryIntermediate1 -> primaryIntermediate2** -> primaryIntermediate3
step 2: Create endpoint1 in Primary using primaryIntermediate3
step 3: Generate **RootCA2 -> remoteIntermediate1 -> remoteIntermediate2** -> remoteIntermediate3
step 4: Create endpoint2 in remote using remoteIntermediate3
step5:  Copy and install ONLY remoteIntermediate3 and endpoint1 in Primary and, primaryIntermediate3 and endpoint2 in remote_. 

Such a configuration gave the below error. Is there a way where we can tell strongswan, that if we give it an intermediate certificate, accept it, even if there is no RootCA and the remaining chain available at both ends.

Thu Feb 10 20:13:56 2022 15[CFG] <ED_USR_IPR_1_0_0_1_65535_IPVX_192-168-180-16_192-168-180-15_3863|286078> no issuer certificate found for "C=x, O=y, CN=intermediate3 CA"

**Reply by tobiasbrunner** [reply]
> I am trying to find out, if there is a way in strongswan to prevent validation of entire certificate chain.

I answered that already.

> even if there is no RootCA and the remaining chain available at both ends.

But you have the root CA installed.
